# Comparison of average values for patients who receive thrombolysis and those that do not

## Aims

Compare feature means for patients who receive thrombolysis and those that do not.

This analysis is for only those patients arriving within 4 hours of known stroke onset, and is on data that has been coded and, where necessary, imputed.The data used in this analysis is the data used for machine learning.

## Load and analyse data

In [1]:
# import libraries
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd

# Set up results DataFrame
results = pd.DataFrame()

# Display entire dataframes
pd.set_option("display.max_rows", 999, "display.max_columns", 150)

# Import data (concenate training and test set used in machine learning)
data = pd.concat([
    pd.read_csv('./../data/10k_training_test/cohort_10000_train.csv'),
    pd.read_csv('./../data/10k_training_test/cohort_10000_test.csv')])

# Add columns for scanned within 4 hours of arrival and onset
data['scan_within_4_hrs_arrival'] = data['S2BrainImagingTime_min'] <= 240
data['scan_within_4_hrs_onset'] = \
    (data['S1OnsetToArrival_min'] + data['S2BrainImagingTime_min']) <= 240
# Convert boolean to integer
data['scan_within_4_hrs_arrival'] *= 1
data['scan_within_4_hrs_onset'] *= 1

In [2]:
data.dtypes

StrokeTeam                                         object
S1AgeOnArrival                                    float64
S1OnsetToArrival_min                              float64
S2RankinBeforeStroke                                int64
Loc                                                 int64
LocQuestions                                      float64
LocCommands                                       float64
BestGaze                                          float64
Visual                                            float64
FacialPalsy                                       float64
MotorArmLeft                                      float64
MotorArmRight                                     float64
MotorLegLeft                                      float64
MotorLegRight                                     float64
LimbAtaxia                                        float64
Sensory                                           float64
BestLanguage                                      float64
Dysarthria    

## Summarise all 4 hour admissions

In [3]:
all_4hr_admissions_summary = data.describe().T
results['all'] = all_4hr_admissions_summary['mean']
all_4hr_admissions_summary

count        mean  \
S1AgeOnArrival                                  88928.0   75.135447   
S1OnsetToArrival_min                            88928.0  111.392722   
S2RankinBeforeStroke                            88928.0    1.059689   
Loc                                             88928.0    0.275178   
LocQuestions                                    88928.0    0.675198   
LocCommands                                     88928.0    0.372133   
BestGaze                                        88928.0    0.354253   
Visual                                          88928.0    0.506016   
FacialPalsy                                     88928.0    0.885717   
MotorArmLeft                                    88928.0    0.869546   
MotorArmRight                                   88928.0    0.823531   
MotorLegLeft                                    88928.0    0.869962   
MotorLegRight                                   88928.0    0.834034   
LimbAtaxia                                      88928.0    0.252339   
Sensory                                         88928.0    0.517745   
BestLanguage                                    88928.0    0.835777   
Dysarthria                                      88928.0    0.746537   
ExtinctionInattention                           88928.0    0.492207   
S2NihssArrival                                  88928.0    9.310172   
S2BrainImagingTime_min                          88928.0  117.212959   
MoreEqual80y_No                                 88928.0    0.576781   
MoreEqual80y_Yes                                88928.0    0.423219   
S1Gender_Female                                 88928.0    0.482188   
S1Gender_Male                                   88928.0    0.517812   
S1Ethnicity_Asian                               88928.0    0.030677   
S1Ethnicity_Black                               88928.0    0.011774   
S1Ethnicity_Mixed                               88928.0    0.003340   
S1Ethnicity_Other                               88928.0    0.071237   
S1Ethnicity_White                               88928.0    0.882973   
S1OnsetInHospital_No                            88928.0    1.000000   
S1OnsetInHospital_Yes                           88928.0    0.000000   
S1OnsetDateType_Best estimate                   88928.0    0.056934   
S1OnsetDateType_Precise                         88928.0    0.897411   
S1OnsetDateType_Stroke during sleep             88928.0    0.045655   
S1OnsetTimeType_Best estimate                   88928.0    0.379599   
S1OnsetTimeType_Not known                       88928.0    0.000000   
S1OnsetTimeType_Precise                         88928.0    0.620401   
S1ArriveByAmbulance_No                          88928.0    0.082381   
S1ArriveByAmbulance_Yes                         88928.0    0.917619   
S1ArriveByAmbulance_missing                     88928.0    0.000000   
S1AdmissionHour_00:00 to 02:59                  88928.0    0.051637   
S1AdmissionHour_03:00 to 05:59                  88928.0    0.029260   
S1AdmissionHour_06:00 to 08:59                  88928.0    0.069258   
S1AdmissionHour_09:00 to 11:59                  88928.0    0.221696   
S1AdmissionHour_12:00 to 14:59                  88928.0    0.202231   
S1AdmissionHour_15:00 to 17:59                  88928.0    0.163649   
S1AdmissionHour_18:00 to 20:59                  88928.0    0.153214   
S1AdmissionHour_21:00 to 23:59                  88928.0    0.109055   
S1AdmissionDay_Friday                           88928.0    0.143802   
S1AdmissionDay_Monday                           88928.0    0.146624   
S1AdmissionDay_Saturday                         88928.0    0.137561   
S1AdmissionDay_Sunday                           88928.0    0.136594   
S1AdmissionDay_Thursday                         88928.0    0.143959   
S1AdmissionDay_Tuesday                          88928.0    0.144746   
S1AdmissionDay_Wednesday                        88928.0    0.146714   
S1AdmissionQuarter_Q1                           88928.0    0.251181   
S1A

## Summarise 4 hour admissions who receive thrombolysis

In [4]:
mask = data['S2Thrombolysis'] == 1
thrombolysis_admissions = data[mask]
thrombolysis_admissions_summary = thrombolysis_admissions.describe().T
results['thrombolysis'] = thrombolysis_admissions_summary['mean']
thrombolysis_admissions_summary

count       mean        std  \
S1AgeOnArrival                                  26257.0  73.092318  13.964103   
S1OnsetToArrival_min                            26257.0  96.968808  45.638922   
S2RankinBeforeStroke                            26257.0   0.727806   1.184651   
Loc                                             26257.0   0.206650   0.509833   
LocQuestions                                    26257.0   0.801653   0.921942   
LocCommands                                     26257.0   0.405263   0.727145   
BestGaze                                        26257.0   0.467266   0.720925   
Visual                                          26257.0   0.678829   0.899433   
FacialPalsy                                     26257.0   1.193548   0.878565   
MotorArmLeft                                    26257.0   1.069886   1.486363   
MotorArmRight                                   26257.0   1.031763   1.478511   
MotorLegLeft                                    26257.0   1.040332   1.420773   
MotorLegRight                                   26257.0   1.011121   1.436217   
LimbAtaxia                                      26257.0   0.291998   0.598986   
Sensory                                         26257.0   0.685036   0.777973   
BestLanguage                                    26257.0   1.070610   1.132034   
Dysarthria                                      26257.0   0.950489   0.755430   
ExtinctionInattention                           26257.0   0.682370   0.818104   
S2NihssArrival                                  26257.0  11.586815   7.016364   
S2BrainImagingTime_min                          26257.0  22.984195  19.822777   
MoreEqual80y_No                                 26257.0   0.643409   0.479002   
MoreEqual80y_Yes                                26257.0   0.356591   0.479002   
S1Gender_Female                                 26257.0   0.460220   0.498425   
S1Gender_Male                                   26257.0   0.539780   0.498425   
S1Ethnicity_Asian                               26257.0   0.031154   0.173736   
S1Ethnicity_Black                               26257.0   0.011654   0.107325   
S1Ethnicity_Mixed                               26257.0   0.003656   0.060357   
S1Ethnicity_Other                               26257.0   0.082873   0.275696   
S1Ethnicity_White                               26257.0   0.870663   0.335579   
S1OnsetInHospital_No                            26257.0   1.000000   0.000000   
S1OnsetInHospital_Yes                           26257.0   0.000000   0.000000   
S1OnsetDateType_Best estimate                   26257.0   0.021975   0.146605   
S1OnsetDateType_Precise                         26257.0   0.972807   0.162648   
S1OnsetDateType_Stroke during sleep             26257.0   0.005218   0.072046   
S1OnsetTimeType_Best estimate                   26257.0   0.180104   0.384282   
S1OnsetTimeType_Not known                       26257.0   0.000000   0.000000   
S1OnsetTimeType_Precise                         26257.0   0.819896   0.384282   
S1ArriveByAmbulance_No                          26257.0   0.059603   0.236755   
S1ArriveByAmbulance_Yes                         26257.0   0.940397   0.236755   
S1ArriveByAmbulance_missing                     26257.0   0.000000   0.000000   
S1AdmissionHour_00:00 to 02:59                  26257.0   0.048292   0.214386   
S1AdmissionHour_03:00 to 05:59                  26257.0   0.017405   0.130777   
S1AdmissionHour_06:00 to 08:59                  26257.0   0.063297   0.243502   
S1AdmissionHour_09:00 to 11:59                  26257.0   0.232548   0.422464   
S1AdmissionHour_12:00 to 14:59                  26257.0   0.215942   0.411482   
S1AdmissionHour_15:00 to 17:59                  26257.0   0.172183   0.377546   
S1AdmissionHour_18:00 to 20:59                  26257.0   0.143885   0.350980   
S1AdmissionHour_21:00 to 23:59                  26257.0   0.106448   0.308416   
S1AdmissionDay_Friday                           26257.0   0.147008   0.354121   


## Summarise 4 hour admissions who do not receive thrombolysis

In [5]:
mask = data['S2Thrombolysis'] == 0
no_thrombolysis_admissions = data[mask]
no_thrombolysis_admissions_summary = no_thrombolysis_admissions.describe().T
results['no_thrombolysis'] = no_thrombolysis_admissions_summary['mean']
no_thrombolysis_admissions_summary

count        mean  \
S1AgeOnArrival                                  62671.0   75.991447   
S1OnsetToArrival_min                            62671.0  117.435848   
S2RankinBeforeStroke                            62671.0    1.198736   
Loc                                             62671.0    0.303889   
LocQuestions                                    62671.0    0.622218   
LocCommands                                     62671.0    0.358252   
BestGaze                                        62671.0    0.306904   
Visual                                          62671.0    0.433614   
FacialPalsy                                     62671.0    0.756746   
MotorArmLeft                                    62671.0    0.785611   
MotorArmRight                                   62671.0    0.736290   
MotorLegLeft                                    62671.0    0.798583   
MotorLegRight                                   62671.0    0.759841   
LimbAtaxia                                      62671.0    0.235723   
Sensory                                         62671.0    0.447655   
BestLanguage                                    62671.0    0.737390   
Dysarthria                                      62671.0    0.661087   
ExtinctionInattention                           62671.0    0.412535   
S2NihssArrival                                  62671.0    8.356337   
S2BrainImagingTime_min                          62671.0  156.691580   
MoreEqual80y_No                                 62671.0    0.548866   
MoreEqual80y_Yes                                62671.0    0.451134   
S1Gender_Female                                 62671.0    0.491392   
S1Gender_Male                                   62671.0    0.508608   
S1Ethnicity_Asian                               62671.0    0.030477   
S1Ethnicity_Black                               62671.0    0.011824   
S1Ethnicity_Mixed                               62671.0    0.003207   
S1Ethnicity_Other                               62671.0    0.066362   
S1Ethnicity_White                               62671.0    0.888130   
S1OnsetInHospital_No                            62671.0    1.000000   
S1OnsetInHospital_Yes                           62671.0    0.000000   
S1OnsetDateType_Best estimate                   62671.0    0.071580   
S1OnsetDateType_Precise                         62671.0    0.865823   
S1OnsetDateType_Stroke during sleep             62671.0    0.062597   
S1OnsetTimeType_Best estimate                   62671.0    0.463181   
S1OnsetTimeType_Not known                       62671.0    0.000000   
S1OnsetTimeType_Precise                         62671.0    0.536819   
S1ArriveByAmbulance_No                          62671.0    0.091924   
S1ArriveByAmbulance_Yes                         62671.0    0.908076   
S1ArriveByAmbulance_missing                     62671.0    0.000000   
S1AdmissionHour_00:00 to 02:59                  62671.0    0.053039   
S1AdmissionHour_03:00 to 05:59                  62671.0    0.034226   
S1AdmissionHour_06:00 to 08:59                  62671.0    0.071756   
S1AdmissionHour_09:00 to 11:59                  62671.0    0.217150   
S1AdmissionHour_12:00 to 14:59                  62671.0    0.196486   
S1AdmissionHour_15:00 to 17:59                  62671.0    0.160074   
S1AdmissionHour_18:00 to 20:59                  62671.0    0.157122   
S1AdmissionHour_21:00 to 23:59                  62671.0    0.110147   
S1AdmissionDay_Friday                           62671.0    0.142458   
S1AdmissionDay_Monday                           62671.0    0.145985   
S1AdmissionDay_Saturday                         62671.0    0.139570   
S1AdmissionDay_Sunday                           62671.0    0.135310   
S1AdmissionDay_Thursday                         62671.0    0.143543   
S1AdmissionDay_Tuesday                          62671.0    0.146016   
S1AdmissionDay_Wednesday                        62671.0    0.147117   
S1AdmissionQuarter_Q1                           62671.0    0.253738   
S1A

## Show summary of all groups

Add ratio of yes/no thrombolysis (and sort by ratio), and save.

In [6]:
results['ratio'] = results['thrombolysis'] / results['no_thrombolysis']
results.sort_values('ratio', inplace=True)
results.to_csv('output/thrombolse_yes_no_means.csv')
results

all  thrombolysis  \
S2StrokeType_Primary Intracerebral Haemorrhage    0.148918      0.000000   
S2StrokeType_missing                              0.001529      0.000000   
S1OnsetDateType_Stroke during sleep               0.045655      0.005218   
AFAnticoagulentDOAC_Yes                           0.031497      0.006056   
S2BrainImagingTime_min                          117.212959     22.984195   
AFAnticoagulentHeparin_Yes                        0.000742      0.000190   
AFAnticoagulent_Yes                               0.121525      0.036943   
S1OnsetDateType_Best estimate                     0.056934      0.021975   
AFAnticoagulentVitK_Yes                           0.019398      0.007846   
S1OnsetTimeType_Best estimate                     0.379599      0.180104   
AFAntiplatelet_No but                             0.029619      0.014091   
S1AdmissionHour_03:00 to 05:59                    0.029260      0.017405   
AFAntiplatelet_No                                 0.135267      0.084549   
AtrialFibrillation_Yes                            0.207235      0.139887   
S2RankinBeforeStroke                              1.059689      0.727806   
S1ArriveByAmbulance_No                            0.082381      0.059603   
S2TIAInLastMonth_Yes                              0.004408      0.003199   
S2TIAInLastMonth_No but                           0.010896      0.007960   
Loc                                               0.275178      0.206650   
StrokeTIA_Yes                                     0.272153      0.207449   
S2TIAInLastMonth_No                               0.074442      0.059222   
MoreEqual80y_Yes                                  0.423219      0.356591   
S1OnsetToArrival_min                            111.392722     96.968808   
CongestiveHeartFailure_Yes                        0.050445      0.044560   
S1AdmissionHour_06:00 to 08:59                    0.069258      0.063297   
Diabetes_Yes                                      0.190963      0.176182   
S1AdmissionHour_00:00 to 02:59                    0.051637      0.048292   
S1AdmissionHour_18:00 to 20:59                    0.153214      0.143885   
S2NewAFDiagnosis_missing                          0.719976      0.686027   
S1Gender_Female                                   0.482188      0.460220   
S1AdmissionDay_Saturday                           0.137561      0.132765   
S1AgeOnArrival                                   75.135447     73.092318   
AFAntiplatelet_Yes                                0.042349      0.041246   
S1AdmissionQuarter_Q1                             0.251181      0.245078   
S1AdmissionHour_21:00 to 23:59                    0.109055      0.106448   
Hypertension_Yes                                  0.534826      0.522146   
S1AdmissionDay_Tuesday                            0.144746      0.141715   
S1AdmissionYear_2016                              0.331021      0.325209   
S1Ethnicity_White                                 0.882973      0.870663   
S1Ethnicity_Black                                 0.011774      0.011654   
AFAnticoagulentVitK_missing                       0.624708      0.618387   
AFAnticoagulentDOAC_missing                       0.624708      0.618387   
AFAnticoagulentHeparin_missing                    0.624708      0.618387   
S1AdmissionQuarter_Q4                             0.252631      0.250143   
S1AdmissionDay_Wednesday                          0.146714      0.145752   
S1OnsetInHospital_No                              1.000000      1.000000   
S1AdmissionYear_2017                              0.335800      0.336291   
AFAnticoagulent_No but                            0.046757      0.046845   
CongestiveHeartFailure_No                         0.949555      0.955440   
S1AdmissionDay_Thursday                           0.143959      0.144952   
S1AdmissionDay_Monday                             0.146624      0.148151   
S1AdmissionQuarter_Q3                             0.244760      0.248353   
S1Ethnicity_Asian          

## Observations

For patients arriving within 4 hours of known stroke onset, compared to those that do not receive thrombolysis, patients who receive thrombolysis:

* Have a confirmed ischaemic stroke 
* Do not have stroke onset during sleep
* Arrive outside of the hours 3am to 6am
* Are younger (mean age 73 vs 76)
* Arrive sooner (mean onset to arrival 97 vs 117 minutes)
* Have higher stroke severity (mean NIHSS 11.6 vs 8.4)
* Are scanned within 4 hours of arrival (100% vs 93%) and 4 hours of onset (99% vs 77%)
* Are more likely to have a precisely determined stroke onset time (97% vs 87%)
* Have arrived by ambulance (94% vs 91%)
* Not have atrial fibrillation (14% vs 24% having AF)
* Not have a history of TIA (21% vs 30% having had TIA)
* Not be on anticoagulant